apiserver/facades/controller/usersecretsdrain/drain.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package usersecretsdrain
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+
+	commonsecrets "github.com/juju/juju/apiserver/common/secrets"
+	apiservererrors "github.com/juju/juju/apiserver/errors"
+	coresecrets "github.com/juju/juju/core/secrets"
+	"github.com/juju/juju/rpc/params"
+	"github.com/juju/juju/secrets"
+	"github.com/juju/juju/secrets/provider"
+)
+
+var logger = loggo.GetLogger("juju.apiserver.usersecretsdrain")
+
+// SecretsDrainAPI is the implementation for the SecretsDrain facade.
+type SecretsDrainAPI struct {
+	*commonsecrets.SecretsDrainAPI
+
+	secretsState SecretsState
+
+	drainConfigGetter   commonsecrets.BackendDrainConfigGetter
+	backendConfigGetter commonsecrets.BackendConfigGetter
+}
+
+// GetSecretBackendConfigs gets the config needed to create a client to secret backends.
+func (s *SecretsDrainAPI) GetSecretBackendConfigs(arg params.SecretBackendArgs) (params.SecretBackendConfigResults, error) {
```

Comment:
Should these methods be in `apiserver/common/secrets` ... the code looks duplicated from the secretsmanager facade?

Reply:
Nah, they are very different. This one is a much simpler version of the facade for agents.
Because the agent facade supports calling by owners and consumers(event CMR), getting secrets by labels, also --peek, --refresh, etc.